src/character.cpp

```diff
@@ -12346,10 +12346,14 @@ stat_mod Character::get_pain_penalty() const
 
     // Prevent negative penalties, there is better ways to give bonuses for pain
     // Also not make character has 0 stats
-    ret.strength = std::clamp( ret.strength, 1, get_str() - 1 );
-    ret.dexterity = std::clamp( ret.dexterity, 1, get_dex() - 1 );
-    ret.intelligence = std::clamp( ret.intelligence, 1, get_int() - 1 );
-    ret.perception = std::clamp( ret.perception, 1, get_per() - 1 );
+    ret.strength = get_str() > 2 ? std::clamp( ret.strength, 1, get_str() - 1 ) :
+                   std::clamp( get_str() - 1, 0, 1 );
```

Comment:
This conflicts with the comment above about preventing 0 stats.

Reply:
`ret.stat` is the amount to be decreased from total stat, so no real conflict I think.
This comment was added in #72955 , I guess the codes and the comment were not matching, and the way my previous PR did was what they realy wanted to add.
@GuardianDll sorry to bother you, do I understand your purpose right?

Reply:
Oh, I misunderstood. Could `get_str()` etc be 0 in the first place? If not, I suggest changing these lines to e.g.
```c++
ret.strength = get_str() >= 2 ? std::clamp( ret.strength, 1, get_str() - 1 ) : 0;
```
The original logic is fine as long as `get_str() >= 2`, and if `get_str() == 1`, `ret.strength` can only be 0 to avoid 0 strength, and if `get_str()` does happen to be less than 1, there is probably something wrong in other parts of the code and we should not modify the stats.

Reply:
get_stat() = max( 0, stat_max + stat_bonus ), so yes it could be 0.
And `ret.strength = get_str() >= 2 ? std::clamp( ret.strength, 1, get_str() - 1 ) : 0;` won't prevent what this PR want to avoid, as `get_str() - 1` can still be 1 and lead to https://github.com/CleverRaven/Cataclysm-DDA/pull/74116#discussion_r1619855171 .

Reply:
Ahh, maybe I should use std::max( 0,  get_str() )